src/error.rs

```diff
@@ -71,6 +82,8 @@ pub enum Error {
     /// A standard error with a string description;
     /// a more specific error should generally be used.
     DefaultError(String),
+    #[cfg(feature = "ssl")]
+    SslError(String),
```

Comment:
I originally wanted to have something better than just a string for SSL errors, but the openssl crate does not do errors very idiomatically, so returning its error types to users would likely just be confusing for users.